driver-core/src/test/unit/com/mongodb/internal/connection/DefaultServerConnectionSpecification.groovy

```diff
@@ -211,7 +213,7 @@ class DefaultServerConnectionSpecification extends Specification {
 
         then:
         1 * executor.executeAsync({ compare(new InsertProtocol(namespace, true, insertRequest), it) }, internalConnection,
-                callback)
+                {expect callback, isTheSameAs(it.wrapped)})
```

Comment:
Had to unwrap `callback` to make the expectation of `executor.executeAsync` work as it was originally.